tasty/src/dotty/tools/tasty/TastyVersion.scala

```diff
@@ -21,4 +21,16 @@ case class TastyVersion(major: Int, minor: Int, experimental: Int) {
     val extra = Option.when(experimental > 0)(this)
     s"stable TASTy from ${min.show} to ${max.show}${extra.fold("")(e => s", or exactly ${e.show}")}"
   }
-}
\ No newline at end of file
+}
+
+object TastyVersion {
+  /** All TASTy versions in stable releases */
+  private val minorVersions: Array[TastyVersion] =
+    Array.tabulate(TastyFormat.MinorVersion + 1)(minor => new TastyVersion(28, minor, 0))
+
+  def apply(major: Int, minor: Int, experimental: Int): TastyVersion = {
+    assert(major == 28, s"TASTy version cache does not yet support major version $major")
```

Comment:
We specifically need to allow any tasty version in order to report errors correctly in `TastyHeaderUnpickler`. we also have tests in TastyHeaderUnpicklerTest for these error messages where I test major versions 27 and 29

Reply:
ok